persistence/soft-index/src/main/java/org/infinispan/persistence/sifs/LogAppender.java

```diff
@@ -1,124 +1,207 @@
-         int currentOffset = 0;
-         while (true) {
-            LogRequest request = queue.pop();
-            if (request != null) {
-               if (request.isClear()) {
-                  logFile.close();
-                  queue.notifyNoWait();
-                  request.pause();
-                  currentOffset = 0;
-                  logFile = fileProvider.getFileForLog();
-                  log.debug("Appending records to " + logFile.fileId);
-                  continue;
-               } else if (request.isStop()) {
-                  queue.notifyNoWait();
-                  break;
-               } else if (request.isPause()) {
-                  queue.notifyNoWait();
-                  request.pause();
-                  continue;
-               }
-               if (currentOffset + request.length() > maxFileSize) {
-                  // switch to next file
-                  logFile.close();
-                  compactor.completeFile(logFile.fileId);
-                  currentOffset = 0;
-                  logFile = fileProvider.getFileForLog();
-                  log.debug("Appending records to " + logFile.fileId);
-               }
-               long seqId = nextSeqId();
-               EntryRecord.writeEntry(logFile.fileChannel, request.getSerializedKey(), request.getSerializedMetadata(),
-                     request.getSerializedInternalMetadata(),
-                     request.getSerializedValue(), seqId, request.getExpiration(), request.getCreated(), request.getLastUsed());
-               int offset = request.getSerializedValue() == null ? ~currentOffset : currentOffset;
-               temporaryTable.set(request.getKey(), logFile.fileId, offset);
-               IndexRequest indexRequest = IndexRequest.update(request.getKey(), raw(request.getSerializedKey()),
-                     logFile.fileId, offset, request.length());
-               request.setIndexRequest(indexRequest);
-               indexQueue.put(indexRequest);
-               currentOffset += request.length();
-            } else {
-               if (syncWrites) {
-                  logFile.fileChannel.force(false);
-               }
-               queue.notifyAndWait();
-            }
+         if (logFile == null) {
+            logFile = fileProvider.getFileForLog();
+            log.debug("Appending records to " + logFile.fileId);
+         }
+         if (request.isClear()) {
+            logFile.close();
+            currentOffset = 0;
+            logFile = null;
+            // TODO: Technically other writes won't be notified if sync write is enabled until
+            // the timeout above
+            delay = AsyncProcessor.create();
+            completeRequest(request);
+            return;
+         } else if (request.isPause()) {
+            delay = AsyncProcessor.create();
+            completeRequest(request);
+            return;
+         } else if (request.isResume()) {
```

Comment:
Nitpick. No need for `else if` with `return`.

Reply:
It is still required. The task is a write based operation otherwise.